test/controllers/projects_controller_test.rb

```diff
@@ -299,24 +299,15 @@ def test_changing_project_name
 
   def test_user_group_save_fail
```

Comment:
The mocha equivalent is:
```ruby
  def test_u
    title = "Bad User Group"
    UserGroup.any_instance.stubs(:save).returns(false)
    post_requires_login(:create, build_params(title, title))
    assert_nil(Project.find_by(title: title))
  end
```

Reply:
I'm not a fan of `any_instance` stubs.  You may be able to do the same thing I'm doing with something like:
```
user_group.stubs(:save),returns(true)
UserGroup.stubs(:new).returns(user_group)
```
But I haven't tried it.  I don't know if you then need to "unstub" UserGroup.new.  Personally, I like the nested context a bit more and since I know that just lasts for as long as that context is running.  I think that also gives it at least a fighting chance of being thread safe, but I don't know the ins and outs of that.  As far as I know I don't think we're currently using multi-threading for tests, but I could imagine getting to the point where it was worthwhile.